dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -726,6 +728,22 @@ public Response getAITopicCandidate(String id) {
         return Response.ok(versionOptional.get().getName()).build();
     }
 
+    @Override
+    public Response getAITopicCandidates() {
+        // Get published entries that don't have any topics
+        List<Entry> entriesWithNoTopics = workflowDAO.getPublishedEntriesWithNoTopics();
```

Comment:
The odd part about this is that this method call will(?) use a ton of memory the first time but very little once we get going. Does it work with the size of the current prod/staging DBs from a performance perspective?

Reply:
A solution with pretty low impact would be to have the query return entry IDs, then retrieve/process/evict them one-by-one in the loop.

Reply:
Should work, the other idea is to do something like a max of 500 at a time. So say if we had 3000 workflows and ran every night, it would take six nights to process everything and give us some time to react/monitor any weirdness. 

We may want to do both actually.

Reply:
I was just looking and the line 738 potentially iterates through all versions in a workflow, depending on whether the default version is set. Given that, if this endpoint is invoked through the LB, there is the risk of the LB closing the connection (I think it's a 2 minute max). Have you timed this against prod data? If it's fast enough anyway, then maybe we're prematurely optimizing. 

And after we process the initial 3,000, there typically won't be enough entries for performance to be an issue.

Reply:
To limit the time consumed, you could also just bail out of the loop and return what was processed after X seconds, where X was low enough to avoid any of the problems.

Reply:
It takes about 10 seconds to run the new topicgenerator command which ends up getting 1084 published entries and writing them to a file. If the number seems less than you thought, it's because I realized the numbers I had in the google doc included both published and unpublished entries -- it's now corrected to only published entries.

Reply:
Seems like it's fine the way it is then.